File: app/channels/chat_channel.rb
# app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
     # Assuming you have a way to identify the chatroom, e.g., by its ID
     stream_from "chatroom_#{params[:chatroom_id]}"
  end

  def speak(data)
     # Broadcast the message to the chatroom
     ActionCable.server.broadcast("chatroom_#{params[:chatroom_id]}", message: data['message'])
  end
 end

File: app/javascript/channels/chat_channel.js
// app/javascript/channels/chat_channel.js
App.chat = App.cable.subscriptions.create({ channel: "ChatChannel", chatroom_id: 1 }, {
  received: function(data) {
     // Update the chat UI with the new message

     var chatBox = document.getElementById('chat-box');
     chatBox.innerHTML += '<p>' + data.message + '</p>';
  }
 });
